### PR TITLE
Fix invalid traceline decode for datapath creation

### LINKED ISSUE #1040: IP Address Decode Out of Bounds Exception
### Describe the bug

I was looking at a log from QNS ([here](https://interop.seemann.io/logs/2020-11-17T08:02/quiche_msquic/handshakecorruption/client/quic.log)) and saw this exception:

```
Invalid TraceLine : timestamp = 08:43:01.967155519, delta = +0.000022000, trace:hostname = client, name = CLOG_DATAPATH_LINUX_C:DatapathCreated, stream.packet.context = { cpu_id = 0 }, stream.event.context = { vpid = 95, vtid = 100 }, event.fields = { arg2 = 0x7F8F48002290, arg3_len = 0, _arg3_length = 0, arg3 = [ ], arg4_len = 28, _arg4_length = 28, arg4 = [ [0] = 2, [1] = 0, [2] = 1, [3] = -69, [4] = -63, [5] = -89, [6] = 100, [7] = 100, [8] = 0, [9] = 0, [10] = 0, [11] = 0, [12] = 0, [13] = 0, [14] = 0, [15] = 0, [16] = 0, [17] = 0, [18] = 0, [19] = 0, [20] = 0, [21] = 0, [22] = 0, [23] = 0, [24] = 0, [25] = 0, [26] = 0, [27] = 0 ] } System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at msquic.clog_config.Types.ADDR(Byte[] value)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at clog2text_lttng.CLogCustomTraceEmittorFactory.Decode(CLogEncodingCLogTypeSearch type, IClogEventArg value, CLogLineMatch traceLine, String& decodedValue) in /home/vsts/work/1/s/src/clogutils/CLogCustomTraceEmittorFactory.cs:line 159
   at clogutils.ConfigFile.CLogConfigurationFile.DecodeUsingCustomDecoder(CLogEncodingCLogTypeSearch node, IClogEventArg value, CLogLineMatch traceLine, String& decodedValue) in /home/vsts/work/1/s/src/clogutils/ConfigFile/CLogConfigurationFile.cs:line 228
   at clogutils.CLogConsoleTrace.DecodeAndTraceToConsole(StreamWriter outputfile, CLogDecodedTraceLine bundle, String errorLine, CLogConfigurationFile config, Dictionary`2 valueBag, EventInformation eventInfo, Boolean showTimeStamp, Boolean showCPUInfo) in /home/vsts/work/1/s/src/clogutils/CLogConsoleTrace.cs:line 89
```

### Expected vs actual behavior

We obviously shouldn't be throwing any exceptions when trying to decode traces. It likely a bug in the decode code.

### PR DESCRIPTION
**Pull request by ThadHouse**
Local path was empty, which is not something we handled previously.

Closes #1040

### CODE REVIEW

File: src/manifest/msquic.clog.cs [hunk 1]
```diff
@@ -16,6 +16,11 @@ public class Types
     {
         public static string ADDR(byte[] value)
         {
+            if (value.Length == 0)
```
Comment (nibanks) [inline src/manifest/msquic.clog.cs:19]: Hmmm... Where do we log a NULL address? This same fix is likely needed for WPA parsing as well.

Comment (ThadHouse) [reply src/manifest/msquic.clog.cs:19]: https://github.com/microsoft/msquic/blob/main/src/platform/datapath_linux.c#L1601

Comment (ThadHouse) [reply src/manifest/msquic.clog.cs:19]: We do the same thing in Windows and Kernel too.

**Review by nibanks — APPROVED** [review]

**Comment by ThadHouse** [comment]
The failure was the bimodal perf issue that needs to be dug into. This doesn't touch any perf code, so merging without that.

**Pull request merged**